public/app/screens/main/sections/devices/device-detail/device-detail.jsx

```diff
@@ -0,0 +1,106 @@
+var React = require('react');
+var Router = require('react-router');
+var Link = Router.Link;
+var styleMixin = require('mixins/style-mixin');
+var roomStore = require('stores/room');
+var DeviceState = require('./../device-state');
+
+var DeviceDetail = React.createClass({
+  mixins: [
+    styleMixin(require('./style.scss')),
+    Router.State],
+
+
+  getInitialState: function getInitialState () {
+    var device = roomStore.getDevice(this.getParams().deviceId);
+    return {
+      device: device,
+      rooms: roomStore.getRooms(),
+      selectedRoom: roomStore.getRoomFromDevice(device)
+    };
+  },
+
+  getRoomDropDown: function getRoomDropDown () {
+    var rooms = this.state.rooms.map(function mapper (room, i) {
+      return (
+        <option key={i} value={room.id}>{room.title}</option>
```

Comment:
I'm fine with not adding a component. Anyway you need a mapping function from the item to its value and title, with or without a component. So it doesn't ease code.